src/Database/Model.php

```diff
@@ -0,0 +1,476 @@
+<?php
+
+namespace Radiate\Database;
+
+use ArrayAccess;
+use JsonSerializable;
+use Radiate\Database\Concerns\HasAttributes;
+use Radiate\Database\Concerns\HasGlobalScopes;
+
+class Model implements ArrayAccess, JsonSerializable
+{
+    use HasAttributes;
+    use HasGlobalScopes;
+
+    /**
+     * The primary key for the model.
+     *
+     * @var string
+     */
+    protected $primaryKey = 'ID';
+
+    /**
+     * The array of booted models.
+     *
+     * @var array
+     */
+    protected static $booted = [];
+
+    /**
+     * Indicates if the model exists.
+     *
+     * @var bool
+     */
+    public $exists = false;
+
+    /**
+     * Indicates if the model was inserted during the current request lifecycle.
+     *
+     * @var bool
+     */
+    public $wasRecentlyCreated = false;
+
+    /**
+     * Create a new Eloquent model instance.
+     *
+     * @param  array  $attributes
+     * @return void
+     */
+    public function __construct(array $attributes = [])
+    {
+        $this->bootIfNotBooted();
+
+        $this->fill($attributes);
+
+        $this->syncOriginal();
```

Comment:
According to Laravel's `Model`, the `syncOriginal` method call should occur before the `fill`.